bond/bond.go

```diff
@@ -189,46 +189,46 @@ func setLinksinNetNs(bondConf *bondingConfig, nspath string, releaseLinks bool)
 	if netNs, err = ns.GetNS(nspath); err != nil {
 		return fmt.Errorf("failed to open netns %q: %v", nspath, err)
 	}
+	defer netNs.Close()
 
 	if curnetNs, err = ns.GetCurrentNS(); err != nil {
 		return fmt.Errorf("failed to get init netns: %v", err)
 	}
 
 	if releaseLinks {
-		if err := netNs.Set(); err != nil {
-			return fmt.Errorf("failed to enter netns %q: %v", netNs, err)
-		}
+		err = move(linkNames, netNs, curnetNs)
+	} else {
+		err = move(linkNames, curnetNs, netNs)
 	}
+	return err
+}
 
-	if len(linkNames) >= 2 { // currently supporting two or more links to one bond
-		for _, linkName := range linkNames {
-			// get interface link in the network namespace
-			link, err := netlink.LinkByName(linkName)
-			if err != nil {
-				return fmt.Errorf("failed to lookup link interface %q: %v", linkName, err)
-			}
-
-			// set link interface down
-			if err = netlink.LinkSetDown(link); err != nil {
-				return fmt.Errorf("failed to down link interface %q: %v", linkName, err)
-			}
+func move(links []string, from ns.NetNS, to ns.NetNS) error {
```

Comment:
let's change the name to something that reflects better that we are moving links from one ns to another

Reply:
changed it :)